source/numerics/time_dependent.cc

```diff
@@ -863,7 +864,7 @@ void TimeStepBase_Tria<dim>::refine_grid (const RefinementData refinement_data)
           if (cell->refine_flag_set())
             previous_cells += (GeometryInfo<dim>::max_children_per_cell-1);
           else if (cell->coarsen_flag_set())
-            previous_cells -= (GeometryInfo<dim>::max_children_per_cell-1) /
+            previous_cells -= (double)(GeometryInfo<dim>::max_children_per_cell-1) /
                               GeometryInfo<dim>::max_children_per_cell;
```

Comment:
How did this work before? We were previously doing integer division, and the result was rounded. But we were dividing `(N-1)/N`, which is zero. You are now dividing `(double(N-1))/N`, which is not zero. It seems to me that the previous approach was wrong.